src/frontends/src/query_frontend.hpp

```diff
@@ -248,9 +268,20 @@ class RequestHandler {
     redis_subscriber_.disconnect();
   }
 
-  void add_application(std::string name, std::vector<std::string> models,
-                       InputType input_type, std::string policy,
-                       std::string default_output, long latency_slo_micros) {
+  void set_linked_models_for_app(std::string name,
+                                 std::vector<std::string> models) {
+    std::unique_lock<std::mutex> l(linked_models_for_apps_mutex_);
+    linked_models_for_apps_[name] = models;
+  }
+
+  std::vector<std::string> get_linked_models_for_app(std::string name) {
```

Comment:
What's the rationale behind storing a copy of `linked_models_for_apps_` rather than querying Redis when a request is made for the data? These kinds of queries are probably infrequent and won't present an excessive load on the database. Unless there's a good reason, we should rely on Redis as a source of truth.